Swing Kit Docs (#1730)
vaadin/docs: articles/tools/swingkit/setup/swing-configuration.asciidoc

---
title: Swing Application Configuration
order: 2
layout: page
---

= Swing Configuration

== Maven Setup

The Swing application will need the following dependency to be added:

[source,xml]
----
<dependency>
    <groupId>com.vaadin</groupId>
    <artifactId>swing-kit-client</artifactId>
    <version>{swing-kit-version}</version>
</dependency>
----

=== Environment Variables

When using MacOS or Linux, the following JVM Parameters must be added to run the application:

[source]
----
--add-opens java.desktop/sun.awt=ALL-UNNAMED
--add-opens java.desktop/sun.lwawt=ALL-UNNAMED
--add-opens java.desktop/sun.lwawt.macosx=ALL-UNNAMED
----

On Windows systems there are no such requirements.

=== Additional Note

> The rendering process uses native files which will be stored in the home folder of the system.
> This file will be downloaded from the Internet, if not already present. This will likely happen silently the first time the Swing application is executed.

Reviewer: The second sentence mentions just one file, so I assume that’s more accurate. Also, I’m not sure what “native files” means, please correct me if that makes a difference to the reader.

```suggestion
The rendering process stores a file in the home folder of the system, which is downloaded from the internet, if not already present.
This will likely happen silently the first time the Swing application is executed.
```

Why “likely”? Don't we know for sure? I would rather avoid this “weasel word” here if possible.
